inst/www/js/blaze.js

```diff
@@ -3,12 +3,31 @@
     Shiny.setInputValue(".clientdata_url_state", value, { priority: "event" });
   };
 
+  var getURLComponents = function() {
+    const params = new URLSearchParams(window.location.search);
+    params.delete('redirect');
+    return {
+      params,
+      pathname: window.location.pathname,
+      hash: window.location.hash
+    };
+  };
+
+  var pathURI = function(redirect, {params, hash} = getURLComponents()) {
+    if (!redirect || redirect == window.location.pathname) {
+      return false;
+    }
+    if (params.toString()) redirect = redirect + '?' + params;
```

Comment:
Because `params` and `hash` are never specified in the calls to `pathURI()` I would prefer the arguments are removed and the variables assigned in the body of the function. 

```suggestion
  var pathURI = function(redirect) {
    if (!redirect || redirect == window.location.pathname) {
      return false
    }
    
    let {params, hash} = getURLComponents();
    
    if (params.toString()) {
      redirect += `?${ params }`;
    }
    
```

Reply:
That makes sense. I think I had used at one point and then refactored but kept it around in case we'd want to provide `params` or `hash`. Which clearly isn't necessary anymore.